src/main/java/com/mg105/user_interface/BattleMenu.java

```diff
@@ -101,91 +142,289 @@ public void start(Stage primaryStage) throws Exception {
+        if (source.equals(nextRound)) {
+            nextRound.setDisable(true);
+
+            moving = presenter.roundStart();
+
+            if (moving == null) { //Battle ended
+                ((Stage) ((Button) source).getScene().getWindow()).close();
+                return;
+            }
+
+            //moving != null
+            if (playerNames[0].equals(moving)) {
+                int[] moveStats = presenter.givenCharacterMoveStats(moving);
+                String[] moveNames = presenter.givenCharacterMoveNames(moving);
+
+                grid.add(moveOne, 2, 4);
+                moveOne.setVisible(true);
+                grid.add(moveTwo, 3, 4);
+                moveTwo.setVisible(true);
+
+                moveOne.setText(moveNames[0] + "\n Hp: " + moveStats[0] + ", Dmg: " + moveStats[1]);
+                moveTwo.setText(moveNames[1] + "\n Hp: " + moveStats[2] + ", Dmg: " + moveStats[3]);
+            } else if (playerNames[1].equals(moving)) {
+                int[] moveStats = presenter.givenCharacterMoveStats(moving);
+                String[] moveNames = presenter.givenCharacterMoveNames(moving);
+
+                grid.add(moveOne, 2, 8);
+                moveOne.setVisible(true);
+                grid.add(moveTwo, 3, 8);
+                moveTwo.setVisible(true);
+
+                moveOne.setText(moveNames[0] + "\n Hp: " + moveStats[0] + ", Dmg: " + moveStats[1]);
+                moveTwo.setText(moveNames[1] + "\n Hp: " + moveStats[2] + ", Dmg: " + moveStats[3]);
+            } else if (playerNames[2].equals(moving)) {
+                int[] moveStats = presenter.givenCharacterMoveStats(moving);
+                String[] moveNames = presenter.givenCharacterMoveNames(moving);
+
+                grid.add(moveOne, 2, 12);
+                moveOne.setVisible(true);
+                grid.add(moveTwo, 3, 12);
+                moveTwo.setVisible(true);
+
+                moveOne.setText(moveNames[0] + "\n Hp: " + moveStats[0] + ", Dmg: " + moveStats[1]);
+                moveTwo.setText(moveNames[1] + "\n Hp: " + moveStats[2] + ", Dmg: " + moveStats[3]);
+            } else if (playerNames[3].equals(moving)) {
+                int[] moveStats = presenter.givenCharacterMoveStats(moving);
+                String[] moveNames = presenter.givenCharacterMoveNames(moving);
+
+                grid.add(moveOne, 2, 16);
+                moveOne.setVisible(true);
+                grid.add(moveTwo, 3, 16);
+                moveTwo.setVisible(true);
+
+                moveOne.setText(moveNames[0] + "\n Hp: " + moveStats[0] + ", Dmg: " + moveStats[1]);
+                moveTwo.setText(moveNames[1] + "\n Hp: " + moveStats[2] + ", Dmg: " + moveStats[3]);
+            } else if(opponentNames[0].equals(moving) || opponentNames[1].equals(moving) ||
+                opponentNames[2].equals(moving) || opponentNames[3].equals(moving)) {
+                //If opponent moved, re-enable nextRound button
+                nextRound.setDisable(false);
```

Comment:
Is there really no better way then the `if-else-if-else-if-else` spam? Surely at this point it merits a refactor. (we can decide to do this later)

Reply:
Noted.